orchestrationSpecs/packages/migration-workflow-templates/src/workflowTemplates/testMigrationWithWorkflowCli.ts

```diff
@@ -0,0 +1,267 @@
+        expression: expr.literal(""),
+        description: "Source cluster endpoint URL"
+    }),
+    sourceAllowInsecure: defineParam({ 
+        expression: expr.literal("true"),
+        description: "Allow insecure connections to source cluster"
+    }),
+    sourceVersion: defineParam({ 
+        expression: expr.literal(""),
+        description: "Source cluster version (example : ES 7.10)"
+    }),
+    
+    // TARGET parameters
+    targetEndpoint: defineParam({ 
+        expression: expr.literal(""),
+        description: "Target cluster endpoint URL"
+    }),
+    targetAllowInsecure: defineParam({ 
+        expression: expr.literal("true"),
+        description: "Allow insecure connections to target cluster"
+    }),
+    targetVersion: defineParam({ 
+        expression: expr.literal(""),
+        description: "Target cluster version (example : OS 2.19)"
+    }),
+    
+    // SNAPSHOT parameters
+    snapshotConfig: defineParam({
+        expression: expr.literal(""),
+        description: "Snapshot repository configuration as JSON object fragment (S3_REPO_CONFIG)"
+    })
+};
+
+const WORKFLOW_CONFIGURE_AND_SUBMIT_SCRIPT = `
+set -e -x
+
+echo "Building and submitting migration workflow..."
+
+# Create migration config JSON
+cat > /tmp/migration_config.json << 'EOF'
+{
+  "skipApprovals": true,
+  "sourceClusters": {
+    "source1": {
+      "endpoint": "{{inputs.parameters.sourceEndpoint}}",
+      "allowInsecure": {{inputs.parameters.sourceAllowInsecure}},
+      "version": "{{inputs.parameters.sourceVersion}}",
+      "snapshotRepo": {{inputs.parameters.snapshotConfig}}
+    }
+  },
+  "targetClusters": {
+    "target1": {
+      "endpoint": "{{inputs.parameters.targetEndpoint}}",
+      "allowInsecure": {{inputs.parameters.targetAllowInsecure}},
+      "version": "{{inputs.parameters.targetVersion}}"
+    }
+  },
+  "migrationConfigs": [
+    {
+      "fromSource": "source1",
+      "toTarget": "target1",
+      "snapshotExtractAndLoadConfigs": [
+        {
+          "snapshotConfig": {
+            "snapshotNameConfig": {
+              "snapshotNamePrefix": "source1snapshot1"
+            }
+          },
+          "migrations": [
+            {
+              "metadataMigrationConfig": {
+                "indexAllowlist": ["*"]
+              },
+              "documentBackfillConfig": {
+                "indexAllowlist": ["*"]
+              }
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+EOF
+
+echo "Migration config contents:"
+cat /tmp/migration_config.json
+
+. /etc/profile.d/venv.sh
+source /.venv/bin/activate
+
+# Clear and load configuration
+echo "Clearing existing workflow configuration..."
+workflow configure clear --confirm
+echo "Loading configuration from JSON..."
+cat /tmp/migration_config.json | workflow configure edit --stdin
+
+ # Submit workflow
+echo "Submitting workflow..."
+WORKFLOW_OUTPUT=$(workflow submit 2>&1)
+echo "Workflow submit output: $WORKFLOW_OUTPUT"
```

Comment:
nit - would be safer to just dump this directly to stdout rather than to a variable